### ISSUE TITLE
Amplify headless publish fails to update hosted frontend

### ISSUE THREAD

**Issue by dwil618** [aws-amplify/amplify-cli#10358]
I am using the amplify publish command in my CI/CD pipeline in headless mode with the --yes parameter:

`amplify publish --yes`

This works fine when there are updates to the backend, but if there are only frontend changes the command does not work as it requests user confirmation:

```
No changes detected

? Do you still want to publish the frontend? (Y/n)
```

I think the problem is coming from [here](https://github.com/aws-amplify/amplify-cli/blob/5217f58c97e2e8d1239f70074b72ba7dc872bfa2/packages/amplify-cli/src/commands/publish.ts#L36), where there does not seem to be a check for the yes flag. 

I assume this is not an intended functionality and should be relatively easy to add this check?

I am using the latest amplify cli (8.1.0), and this can be reproduced if required by creating and hosting/publishing an amplify app, then trying to re-publish in headless mode without any changes to the backend.

**Comment by josefaidt** [comment]
Hey @dwil618 :wave: thanks for raising this! We'll be taking a look at this shortly however in the mean time this seems related to a previously-closed bug https://github.com/aws-amplify/amplify-cli/issues/10164

**Comment by dwil618** [comment]
Thanks very much for sharing this, I'll try the workaround with `amplify publish --yes --force` to hopefully unblock us for now, but this seems like it still ought to work without the `--force`?

**Comment by josefaidt** [comment]
Hey @dwil618 I was able to successfully reproduce this behavior, and it appears to be a regression. Using `--force` along with `--yes` should be a sufficient workaround for the mean time, but yes it should work without the force flag.

**Comment by akshbhu** [comment]
Closing in favour of #10774
